src/main/kotlin/org/rust/ide/inspections/RsWrongGenericArgumentsOrderInspection.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import org.rust.ide.annotator.fixes.EncloseExprInBracesFix
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.utils.RsDiagnostic
+import org.rust.lang.utils.addToHolder
+
+/**
+ * Inspection that detects the E0747 error.
+ */
+class RsWrongGenericArgumentsOrderInspection : RsLocalInspectionTool() {
+    override fun buildVisitor(holder: RsProblemsHolder, isOnTheFly: Boolean): RsVisitor =
+        object : RsVisitor() {
+            override fun visitBaseType(type: RsBaseType) {
+                if (!isPathValid(type.path)) return
+                checkGenericArguments(holder, type)
+            }
+
+            override fun visitTraitRef(trait: RsTraitRef) {
+                if (!isPathValid(trait.path)) return
+                checkGenericArguments(holder, trait)
+            }
+
+            override fun visitCallExpr(element: RsCallExpr) = checkGenericArguments(holder, element)
```

Comment:
These 3 options can be unified to `visitPath(RsPath)`. This also takes into account path exprs liks this
```rust
let a = foo::<T>;
a();
```

As I see, this is also actual for `RsWrongGenericArgumentsNumberInspection`